### ISSUE TITLE
pki_private_key_from_base64:  Parsing private key: error:0906D064:PEM routines:PEM_read_bio:bad base64 decode [ERR]: LN: Failed to set hostkey "genkey" (/tmp/pzQmpi).

### ISSUE THREAD

**Issue by shcm888** [CESNET/Netopeer2#1523]
Hello, I used netopeer2 cross compilation to run on the arm board. Sysrepo and netopee2 ran normally and started successfully. Netopeer2 also follows the instructions to set up. sh and merge_ Hostkey.sh, and merge_ The config.sh files were manually created, but now after startup, when using netopeer2 cli on arm to connect to netopeer2, the following error message appears:
netopeer2-cli
```
load_config: No saved configuration.
> verb 3
> connect
nc DEBUG: nc_sock_connect(localhost, 830, -1, -1)
nc VERBOSE: Trying to connect via IPv4 to 127.0.0.1:830.
nc VERBOSE: Successfully connected to localhost:830 over IPv4.
[2024/01/08 06:39:33.080225, 3] ssh_config_parse_file:  Reading configuration data from /etc/ssh/ssh_config
[2024/01/08 06:39:33.081201, 2] ssh_connect:  libssh 0.9.7 (c) 2003-2021 Aris Adamantiadis, Andreas Schneider and libssh contributors. Distributed under the LGPL, please refer to COPYING file for information about your rights, using threading threads_pthread
[2024/01/08 06:39:33.081405, 2] ssh_connect:  Socket connecting, now waiting for the callbacks to work
[2024/01/08 06:39:33.081721, 3] ssh_socket_pollcallback:  Received POLLOUT in connecting state
[2024/01/08 06:39:33.085239, 1] socket_callback_connected:  Socket connection callback: 1 (0)
[2024/01/08 06:39:33.087711, 3] ssh_client_connection_callback:  session_state=2
[2024/01/08 06:39:33.090941, 3] ssh_socket_unbuffered_write:  Enabling POLLOUT for socket
[2024/01/08 06:39:33.093521, 3] ssh_connect:  current state : 2
[2024/01/08 06:39:33.095673, 3] ssh_connect:  current state : 2
[2024/01/08 06:39:33.097410, 3] ssh_connect:  current state : 2
[2024/01/08 06:39:33.110894, 3] ssh_connect:  current state : 2
[2024/01/08 06:39:33.112453, 3] ssh_connect:  current state : 2
[2024/01/08 06:39:33.113210, 3] ssh_connect:  current state : 2
[2024/01/08 06:39:33.114264, 3] ssh_connect:  current state : 2
[2024/01/08 06:39:33.116084, 3] ssh_connect:  current state : 2
[2024/01/08 06:39:33.117423, 3] ssh_connect:  current state : 2
[2024/01/08 06:39:33.118230, 3] ssh_connect:  current state : 2
[2024/01/08 06:39:33.119626, 3] ssh_connect:  current state : 2
[2024/01/08 06:39:33.120780, 3] ssh_connect:  current state : 2
[2024/01/08 06:39:33.122059, 3] ssh_connect:  current state : 2
[2024/01/08 06:39:33.124796, 3] ssh_connect:  current state : 2
[2024/01/08 06:39:33.125677, 3] ssh_connect:  current state : 2
[2024/01/08 06:39:33.128583, 1] ssh_socket_exception_callback:  Socket exception callback: 2 (104)
[2024/01/08 06:39:33.129799, 1] ssh_socket_exception_callback:  Socket error: Connection reset by peer
[2024/01/08 06:39:33.131008, 3] ssh_client_connection_callback:  session_state=9
[2024/01/08 06:39:33.132431, 3] ssh_connect:  current state : 9
nc ERROR: Starting the SSH session failed (Socket error: Connection reset by peer).
nc DEBUG: Error code 2.
cmd_connect: Connecting to the localhost:830 as user "root" failed.
```
Netopeer2 server runtime prompts the following error: pki_ Private_ Key_ From_ Base64: Parsing private key: error: 0906D064: PEM routes: PEM_ Read_ Bio: bad base64 decode, as follows:
```
~#:netopeer2-server -d -v3
[INF]: SR: Connection 18 created.
[INF]: SR: Session 7 (user "root", CID 18) created.
[INF]: SR: Triggering "ietf-subscribed-notifications" "done" event on enabled data.
[INF]: SR: Triggering "ietf-netconf-server" "done" event on enabled data.
[INF]: SR: Triggering "ietf-netconf-server" "done" event on enabled data.
[INF]: SR: Triggering "ietf-netconf-server" "done" event on enabled data.
[INF]: LN: Listening on 0.0.0.0:830 for SSH connections.
[INF]: SR: Triggering "ietf-netconf-server" "done" event on enabled data.
[INF]: SR: Triggering "ietf-netconf-server" "done" event on enabled data.
[INF]: SR: Triggering "ietf-netconf-server" "done" event on enabled data.
[INF]: SR: Triggering "ietf-netconf-server" "done" event on enabled data.
[INF]: SR: Triggering "ietf-netconf-server" "done" event on enabled data.
[INF]: SR: Triggering "ietf-netconf-server" "done" event on enabled data.
[INF]: SR: Triggering "ietf-netconf-server" "done" event on enabled data.
[INF]: SR: Triggering "ietf-netconf-server" "done" event on enabled data.
[INF]: SR: Triggering "ietf-netconf-server" "done" event on enabled data.
[INF]: SR: Triggering "ietf-netconf-server" "done" event on enabled data.
[INF]: SR: Triggering "ietf-netconf-server" "done" event on enabled data.
[INF]: SR: Triggering "ietf-netconf-server" "done" event on enabled data.
[INF]: SR: Triggering "ietf-netconf-server" "done" event on enabled data.
[INF]: SR: Triggering "ietf-netconf-server" "done" event on enabled data.
[INF]: SR: Triggering "ietf-netconf-server" "done" event on enabled data.
[INF]: SR: Triggering "ietf-netconf-server" "done" event on enabled data.
[INF]: SR: Triggering "ietf-netconf-server" "done" event on enabled data.
[INF]: SR: Triggering "ietf-netconf-server" "done" event on enabled data.
[INF]: SR: Triggering "ietf-netconf-server" "done" event on enabled data.
[INF]: SR: Triggering "ietf-netconf-server" "done" event on enabled data.
[INF]: SR: Triggering "ietf-keystore" "done" event on enabled data.
[INF]: SR: Triggering "ietf-truststore" "done" event on enabled data.
[INF]: SR: Triggering "ietf-netconf-acm" "done" event on enabled data.
[INF]: SR: Triggering "ietf-netconf-acm" "done" event on enabled data.
[INF]: SR: Triggering "ietf-netconf-acm" "done" event on enabled data.
[INF]: SR: Triggering "ietf-netconf-acm" "done" event on enabled data.
[INF]: LN: Accepted a connection on 0.0.0.0:830 from 127.0.0.1:49772.
[INF]: SR: Session 38 (user "root", CID 18) created.
[2024/01/08 06:39:33.109232, 1] pki_private_key_from_base64:  Parsing private key: error:0906D064:PEM routines:PEM_read_bio:bad base64 decode
[ERR]: LN: Failed to set hostkey "genkey" (/tmp/rVMnui).
```
-------------------------------
netopeer2-server 2.1.71 、  sysrepo v2.2.105 、 libssh 0.9.7、  libnetconf2 v2.1.37  、 libyang v2.1.111  
-------------------------------
https://github.com/CESNET/netopeer2/issues/534  
I have also looked at the issue with this address, but it is of no use.

May I ask how to resolve this error? Could you please provide some suggestions? Thank you！

**Comment by michalvasko** [comment]
It seems the value of the private key is invalid, I cannot tell you why. You can try to set your own keys into the configuration, not sure what other advice to give you.

**Comment by shcm888** [comment]
Thank you very much for your reply. After changing the version of OpenSSL, the issue was resolved (OpenSSL 1.0.1k was changed to OpenSSL 1.0.2k)
